src/libcrun/linux.c

```diff
@@ -1947,6 +1947,67 @@ do_mounts (libcrun_container_t *container, int rootfsfd, const char *rootfs, con
   return 0;
 }
 
+/*
+ * libcrun_container_do_bind_mount
+ *
+ *  Allows external plugins and handlers to perform bind `mounts` on container.
+ *  returns: 0 if successful anything else states `error` and configures `err` with relevent error.
+ */
+int
+libcrun_container_do_bind_mount (libcrun_container_t *container, char *mount_source, char *mount_destination, char **mount_options, size_t mount_options_len, libcrun_error_t *err)
+{
+  int ret, rootfsfd;
+  size_t rootfs_len = get_private_data (container)->rootfs_len;
+
+  const char *target = consume_slashes (mount_destination);
+  cleanup_free char *data = NULL;
+  unsigned long flags = 0;
+  unsigned long extra_flags = 0;
+  cleanup_close int targetfd = -1;
+  int is_dir = 1;
+  uint64_t rec_clear = 0;
+  uint64_t rec_set = 0;
+  const char *rootfs = get_private_data (container)->rootfs;
+  rootfsfd = get_private_data (container)->rootfsfd;
```

Comment:
if rootfsfd is not a valid fd (`rootfsfd < 0`), let's return an error

Reply:
fixed